providers/implementations/keymgmt/rsa_kmgmt.c

```diff
@@ -630,6 +641,11 @@ static void *rsa_gen(void *genctx, OSSL_CALLBACK *osslcb, void *cbarg)
 
     rsa = rsa_tmp;
     rsa_tmp = NULL;
+#ifdef FIPS_MODULE
+    /* Pairwise consistency test */
+    if (do_rsa_pct(gctx->prov_rsa_ctx, "sha256", rsa) != 1)
```

Comment:
ossl_rsa_sp800_56b_generate_key() already does a PCT?

Reply:
Agreed, and I have an impression that in FIPS mode we do it twice, once in ossl_rsa_sp800_56b_generate_key and second time in the last line of the snippet https://github.com/openssl/openssl/blob/ebce766bb82b472eb3b796c5ee5ee2c19beb71f5/crypto/rsa/rsa_gen.c#L430-L451

Reply:
Also as our lab requires us all PCTs should be done for the certified API (DigestSign/DigestVerify) so these low-level functions are used in our patch for RSA and EC